### PR TITLE
Added Api for Deal, Organization, Settings, Tag, Segment

### PR DESCRIPTION
**Pull request by ravimadan86**
Added Api for Deal, Organization, Settings, Tag, Segment

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ composer.phar
 .buildpath
 .project
 .settings
+composer.lock
```
Comment (EVODelavega) [inline .gitignore:11]: Why would you *not* track the lock file? If there are dependencies on a specific version of some other lib, then surely the lock file should ensure that this lib has that dependency available to it. Otherwise updating the dependencies might break this library.

File: README.md [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
+composer require ravimadan/activecampaign-api-php
+
+```
+
+or
+
+```json
     "require": {
-        "EVODelavega/activecampaign-api-php": "dev-master"
+        "ravimadan/activecampaign-api-php": "dev-master"
```
Comment (EVODelavega) [inline README.md:17]: This shouldn't change if the idea is to merge this into this repo

File: composer.json [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-    "name": "evodelavega/activecampaign-api-php",
+    "name": "ravimadan/activecampaign-api-php",
     "description": "An alternative, more PSR-compliant AC API wrapper",
     "license": "MIT",
     "keywords": ["activecampaign", "email-marketing", "newsletter", "marketing-automation", "subscribe", "forms", "emails", "automation"],
     "authors": [
         {
-            "name": "Elias Van Ootegem",
-            "role": "Developer"
+            "name": "Ravi Madan",
```
Comment (EVODelavega) [inline composer.json:8]: Rather than replace, maybe add a second author. I'm fine with crediting anyone and everyone who worked on this, myself included. I'm not working with PHP anymore though, so if you'd prefer, I'm more than happy giving you ownership over this repo

File: src/AC/Deal.php [hunk 4]
```diff
@@ -0,0 +1,128 @@
+<?php
+namespace AC;
+
+class Deal extends ActiveCampaign {
+	
+	function add($params, $post_data) {
```
Comment (EVODelavega) [inline src/AC/Deal.php:6]: Do specify visibility of methods (use `public function` instead of implicit public `function`). Try to keep code PSR-compliant: 

* Visibility MUST be declared on all properties and methods; abstract and final MUST be declared before the visibility; static MUST be declared after the visibility.
* Method names MUST be declared in `camelCase`.

There's something about consistent variable naming. I always used `camelCase` for both variables and method names. Method names are a PSR thing. IIRC, there's something in the coding standards about being consistent WRT to variable names (`$request_url` should therefore be `$requestUrl`). Given that I've been writing golang for a couple of years now, I'd write `$requestURL` now anyway.

Just have a read through https://www.php-fig.org/

File: src/AC/Deal.php [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+	function tasktype_delete($params, $post_data) {
+		$request_url = "{$this->url}&api_action=deal_tasktype_delete&api_output={$this->output}";
+		$response = $this->curl($request_url, $post_data);
+		return $response;
+	}
+
+}
+
+?>
```
Comment (EVODelavega) [inline src/AC/Deal.php:128]: No closing tags (PSR)

File: src/AC/Deal.php [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+	}
+
+	function tasktype_edit($params, $post_data) {
+		$request_url = "{$this->url}&api_action=deal_tasktype_edit&api_output={$this->output}";
+		$response = $this->curl($request_url, $post_data);
+		return $response;
+	}
+
+	function tasktype_delete($params, $post_data) {
+		$request_url = "{$this->url}&api_action=deal_tasktype_delete&api_output={$this->output}";
```
Comment (EVODelavega) [inline src/AC/Deal.php:121]: All these methods could be written as one-liners:

```php
    return $this->curl($this->url."&api_action=deal_tasktype_delete&api_output=". $this->output, $postData);
```

File: src/AC/Deal.php [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+class Deal extends ActiveCampaign {
+	
+	function add($params, $post_data) {
+		$request_url = "{$this->url}&api_action=deal_add&api_output={$this->output}";
+		$response = $this->curl($request_url, $post_data);
+		return $response;
+	}
+
+	function edit($params, $post_data) {
```
Comment (EVODelavega) [inline src/AC/Deal.php:12]: AFAIK, the arguments can be type-hinted to enforce `array` arguments to be passed

File: src/AC/Organization.php [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+    function list_($params, $post_data) {
+        $request_url = "{$this->url}&api_action=organization_list&api_output={$this->output}";
+        $response = $this->curl($request_url);
+        return $response;
+    }
+
+}
+
+?>
```
Comment (EVODelavega) [inline src/AC/Organization.php:14]: PSR, remove closing `?>`

File: src/AC/Segment.php [hunk 9]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+class Segment extends ActiveCampaign {
```
Comment (EVODelavega) [inline src/AC/Segment.php:3]: PSR: class opening `{` should go on the next line

**Review by EVODelavega — CHANGES_REQUESTED** [review]
I've not written PHP in a long time. I'm more than happy to transfer ownership of the repo. I did look at the PR a bit more closely. There's a couple of issues I'd recommend you address. Mainly to do with following the coding standards.

Other issues:

* Adding `composer.lock` to the gitignore file is a bad idea
* The composer.json file I added specifies the MIT license. Removing the author, and replacing it with your own details is, I think, not allowed. Just add yourself as another author, but don't remove what was originally there
* If you don't want to take ownership iof this repo, the name of the lib should *not* be changed in the composer.json file

**Comment by EVODelavega** [comment]
So TL;DR:

I'm fine merging this in, provided you address some of the issues I pointed out. The ones I think you are, in fact, legally required to make are:

* Removing of original name and author (this violates the MIT license). At the time I forked this, I reached out to someone at Active Campaign to ask if they were happy with my using the MIT license, let's not violate the agreed upon license.
* Removing the `composer.lock` file from the .gitignore file
* Specifying visibility on methods. This was part of PSR-2, which has been around for years.
* Remove closing `?>`

Optionally change variable and property names.

Should you want to take ownership of this repo, then I'll be happy to transfer it, but again have to ask you to _add_ yourself as author, rather than replace my details with yours. The MIT license is very liberal. You can add what you like, add your own credentials, but you can't remove any of the existing credits.

File: .gitignore (hunk 1 above)
Comment (ravimadan86) [reply .gitignore:11]: the file got committed by mistake and i too never commit it. I will make the change

File: README.md (hunk 2 above)
Comment (ravimadan86) [reply README.md:17]: I will remove it. I wanted to put another repo with credit to you and active campaign but not intended to commit in your repo. I ll make the desired changes
